apps/common/components/Input.tsx

```diff
@@ -0,0 +1,53 @@
+import type {ReactElement} from 'react';
+
+type TAmountInputProps = {
+	value: string;
+	onChange?: (value: string) => void;
+	label?: string;
+	placeholder?: string;
+	legend?: string;
+	error?: string;
+	disabled?: boolean;
+	loading?: boolean;
+}
+
+function Input({
+	value,
+	onChange,
+	label,
+	placeholder,
+	legend,
+	error,
+	disabled,
+	loading
+}: TAmountInputProps): ReactElement {
```

Comment:
Just for readability, when it gets to these tower of pisa props, wdyt we just inline the props like so

```suggestion
function Input(props: TAmountInputProps): ReactElement {
	const {value, onChange, label, placeholder, legend, error, disabled, loading} = props;
```